js/tests/unit/button.js

```diff
@@ -110,6 +110,18 @@ $(function () {
     assert.ok($btn.hasClass('active'), 'btn has class active')
   })
 
+  QUnit.test('should prevent toggle active when disabled btn children are clicked', function (assert) {
+    assert.expect(2)
+    var $btn = $('<button class="btn disabled" data-toggle="button">mdo</button>')
```

Comment:
you are correct. I will update this test.

Though I also have am not sure if this is the correct way to solve this PR. I will continue discussion in #16703.